### ISSUE TITLE
Sound output: observing a delay between speech followed by sound output

### ISSUE THREAD

**Issue by josephsl** [nvaccess/nvda#17517]
Likely a critical regression:

### Steps to reproduce:
Prerequisite: NVDA alpha-34772 and alpha-34782 installers (or installed):

First case: searching in Start menu ("play sounds when suggestions appear" must be enabled):

1. Open Start menu (Windows ley).
2. Start typing text.
3. NVDA will announce the top result and suggestions open sound.

Expected: suggestions open sound followed by top suggestion
Actual: delay between top suggestion announcement and suggestions open sound

Second case: toggling focus and browse modes ("audio indication of focus and browse modes" must be enabled):

1. Open search engines such as Google on a web browser.
2. Press Tab to move through search field and other elements.
3. Alternatively, press NVDA+Space to toggle focus and browse modes.

Expected: focus/browse mode sounds are heard while pressing Tab to navigate elements
Actual: elements are announced first, followed by focus/browse mode toggle sonds
 
### Actual behavior:
There is a delay between speech output and NVDA sonds

### Expected behavior:
No delay between speech and sond, or when searching for items in Start menu (first case), suggestions open sound is heard prior to speech output

#### NVDA installed/portable/running from source:
Installed

#### NVDA version:
alpha-34782,82fefa5a

#### Windows version:
Windows 11 24H2 Insider dev (build 26120.2510)

#### Name and version of other software in use when reproducing the issue:
Microsoft Edge 132.0.2957.41

#### Other information about your system:
None

#### Does the issue still occur after restarting your computer?
Yes

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
No issues in alpha-34772,c3e899a8

#### If NVDA add-ons are disabled, is your problem still occurring?
Yes

#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Not applicable

### Likely cause:
Since alpha-34782 comes with sound changes such as no more logo sound when starting the launcher/installer and removal of WinMM support, I think a regression was introduced in a recent commit. IN the meantime, I have downgraded to alpha-34772 as I need to work on important tasks requiring timely announcement of browse/focus mode toggles in web browsers.

Thanks.

**Comment by Adriani90** [comment]
cc: @SaschaCowley

**Comment by josephsl** [comment]
Hi,

Turns out this is resolved somewhat in alpha-34843,2477780d possibly as a side-effect of removing "buffered" argument in nvwave.WavePlayer constructor. However, some add-ons (some synthesizer add-ons included) still set "buffered" to True despite its removal, and deprecation warning is not set for the now removed parameter in earlier NVDA releases. For now, I'll close it (please reopen or create a new issue if folks discover other sound delay issues after WASAPI is widely adopted).

Thanks.
